### PR TITLE
Improve Containerfile

### PR DESCRIPTION
**Pull request by dcermak**
- fix `set -eux pipefail` to `set -euxo pipefail`, without the `o`, `pipefail` does nothing
- remove pointless `cd $NOTEBOOKS`
- define $NOTEBOOKS as a volume

### CODE REVIEW

File: linux/start/jupyter/adoc/gs_sles_jupyter-jupyterlab_container.adoc [hunk 1]
```diff
@@ -547,19 +547,18 @@ WORKDIR /home/jupyter
 COPY --chown=1000:1000 requirements.txt .
 
 # install JuypterLab
-RUN set -eux pipefail; \
+RUN set -euxo pipefail; \
```
Comment (tlssuse) [inline linux/start/jupyter/adoc/gs_sles_jupyter-jupyterlab_container.adoc:550]: Yes, this change should be made.

File: linux/start/jupyter/adoc/gs_sles_jupyter-jupyterlab_container.adoc [hunk 2]
```diff
@@ -547,19 +547,18 @@ WORKDIR /home/jupyter
...
     pip install --upgrade pip; \
     pip install -r requirements.txt; \
     pip cache purge;
 
 # expose JupyterLab web port
 EXPOSE 8888/tcp 
 
-# change to notebooks directory
-RUN cd $NOTEBOOKS
+VOLUME $NOTEBOOKS
```
Comment (tlssuse) [inline linux/start/jupyter/adoc/gs_sles_jupyter-jupyterlab_container.adoc:558]: The guide uses a "launch script" to handle volume mapping when the container is run.
Adding the VOLUME command to the Container file is an unnecessary complication.

The original lines 558 and 559 are irrelevant and can be removed.

File: linux/start/jupyter/adoc/gs_sles_jupyter-jupyterlab_container.adoc [hunk 3]
```diff
@@ -547,19 +547,18 @@ WORKDIR /home/jupyter
...
 
 # change to notebooks directory and launch JupyterLab
-CMD cd $NOTEBOOKS; jupyter-lab --ip=* --port=8888 --no-browser --notebook-dir=$NOTEBOOKS
+CMD jupyter-lab --ip=* --port=8888 --no-browser --notebook-dir=$NOTEBOOKS
```
Comment (tlssuse) [inline linux/start/jupyter/adoc/gs_sles_jupyter-jupyterlab_container.adoc:561]: `cd $NOTEBOOKS; ` is also irrelevant.
So, this suggested change should be accepted.

**Review by tlssuse — CHANGES_REQUESTED** [review]
Good catch on `RUN set -eux**o** pipefail;`.
Let's make this change.

You are correct that changing to the $NOTEBOOKS directory (lines 558-559) are superfluous.
The guide uses a "launch script" to handle volume mapping when the container is run.
So, adding the VOLUME declaration to Containerfile is an unneeded complication.
I recommend deleting the directory change lines (558-559 of the original file) and NOT adding the `VOLUME $NOTEBOOKS` line.

Likewise, I agree that `cd $NOTEBOOKS; ` is unnecessary in the final `CMD` line.

**Comment by chabowski** [comment]
@tlssuse you are requesting one more change from @dcermak, correct? 
I will merge and push/publish the doc after this is done. Thanks!!

**Comment by chabowski** [comment]
All requested changes have been implemented with PR #99. Closing this PR without merging.

**Pull request closed without merging**